tinygrad/tensor.py

```diff
@@ -988,7 +988,27 @@ def cast(self, dtype:DType) -> Tensor:
     if self.dtype == dtypes.bfloat16: return self.bitcast(dtypes.uint16).cast(dtypes.uint32).mul(1<<16).bitcast(dtypes.float32).cast(dtype)
     return mlops.Cast.apply(self, dtype=dtype)
   def bitcast(self, dtype:DType) -> Tensor:
-    assert self.dtype.itemsize == dtype.itemsize, "can't bitcast mismatched dtype itemsizes"
+    if self.dtype.itemsize != dtype.itemsize:
+      assert self.shape != (), "bitcasting between types of different sizes is not allowed for scalar tensors"
+      to_uint = {1: dtypes.uint8, 2: dtypes.uint16, 4: dtypes.uint32, 8: dtypes.uint64}
+      if self.dtype.itemsize > dtype.itemsize:
+        repeat = self.dtype.itemsize // dtype.itemsize
+        as_repeated_uint = self.repeat([1] * (self.ndim - 1) + [repeat]).reshape(self.shape[:-1] + (repeat, -1))\
+          .transpose(-2, -1).flatten(-2).bitcast(to_uint[self.dtype.itemsize])
+        idxs = Tensor.arange(repeat).repeat([self.shape[-1]])
+        bits = 8 * dtype.itemsize
+        shift = (2 ** (bits * idxs)).cast(to_uint[self.dtype.itemsize])
+        def div(t1: Tensor, t2: Tensor): return mlops.Div.apply(t1, t2)
```

Comment:
I had to add this temporarily to focus on implementing bitcast because `.div` sometimes returns incorrect results when used against unsigned data types.